src/coap_openssl.c

```diff
@@ -491,7 +491,7 @@ coap_dtls_psk_client_callback(
   }
   else {
     /* Reduce to match */
-    max_identity_len = psk_identity->length;
+    max_identity_len = (unsigned int)psk_identity->length;
```

Comment:
Instead of doing the typecasts here, I suggest changing the data type of `max_identity_len` to `size_t`. This would need changing %d to %zu in the log message, though.

Reply:
Unfortunately, `max_identity_len` is passed in as a parameter.  I guess that I could set up a local variable to use.

Reply:
Updated parameter list to be more consistent/intuative